presto-main/src/main/java/com/facebook/presto/cost/ComparisonStatsCalculator.java

```diff
@@ -188,27 +191,41 @@ private static PlanNodeStatsEstimate symbolToSymbolEquality(PlanNodeStatsEstimat
         double filterFactor = 1 * leftFilterFactor * rightFilterFactor / max(leftNdvInRange, rightNdvInRange, 1);
         double retainedNdv = min(leftNdvInRange, rightNdvInRange);
 
-        SymbolStatsEstimate newLeftStats = buildFrom(leftStats)
-                .setNullsFraction(0)
-                .setStatisticsRange(intersect)
-                .setDistinctValuesCount(retainedNdv)
-                .build();
-        SymbolStatsEstimate newRightStats = buildFrom(rightStats)
+        PlanNodeStatsEstimate.Builder estimate = PlanNodeStatsEstimate.buildFrom(inputStatistics)
```

Comment:
For expressionToLiteral you do inputStatistics.mapOutputRowCount, and then get a PlanNodeStatsEstimate that gets used.  Here you create a builder that you keep adding to and build later.  Is there a reason for the difference?

Reply:
Just incidental. From the fact that builder and mutating methods on PlanNodeStatsEstimate itself allow to achieve the same goal in two ways. I will update `expressionToLiteral` to use the builder.